sklearn_pandas/dataframe_mapper.py

```diff
@@ -257,13 +267,21 @@ def get_names(self, columns, transformer, x, alias=None):
             # Otherwise use the only estimator present
             else:
                 names = _get_feature_names(transformer)
+
             if names is not None and len(names) == num_cols:
-                return ['%s_%s' % (name, o) for o in names]
-            # otherwise, return name concatenated with '_1', '_2', etc.
+                output = [f"{name}_{o}" for o in names]
+                # otherwise, return name concatenated with '_1', '_2', etc.
             else:
-                return [name + '_' + str(o) for o in range(num_cols)]
+                output = [name + '_' + str(o) for o in range(num_cols)]
         else:
-            return [name]
+            output = [name]
+
+        if not prefix and not suffix:
+            return output
+
+        prefix = prefix or ''
```

Comment:
This prefix/suffix feature should be explained in the README.

Reply:
Prefix and suffix already default to the empty string in the function signature. Can we skip repeating the default here?

Reply:
okie